staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/matching/matching.go

```diff
@@ -86,6 +89,10 @@ func (m *Matcher) Matches(attr admission.Attributes, o admission.ObjectInterface
 		return false, nil
 	}
 
+	if isBinding && matchResources.ResourceRules == nil {
```

Comment:
For simplicity's sake, Bindings and Definitions should have the same semantics for their match criteria... I'd treat undefined as an implicit `"*/*"`.

In general, match criteria should have the semantics "undefined => match everything", since that makes adding new fields trivial from a backwards compatibility perspective, and consistency makes the semantics easier to reason about.

Reply:
xref https://github.com/kubernetes/kubernetes/pull/113314#discussion_r1013334637

I'm a bit wary of policy definitions matching everything implicitly. I've thought about adding a validation rule that policy definitions must have at least one resourceRule, so if a policy definition is going to match everything, it would have to wildcard explicitly.  I'm more comfortable with bindings wild-carding implicitly.

If we handle it with a validation rule we don't need to pass isBinding in here..

Reply:
I think having a validation rule is reasonable. To be sure, this is alpha, right?

Reply:
Note: agreement was reached to have the semantics "undefined => match everything" and add validation rule against policy definitions. Thanks for everyone's input!